main.cpp

```diff
@@ -131,268 +172,271 @@ struct CarWash
-*/
+// UDT 1
 
 struct Cellphone
 {
-// 5 properties: 
-    // 1) Number of camera lens (int)
     int numCameraLens = 2;
-    //     2) Manufacturer (std::string)
     std::string manufacaturer = "Apple";
-    //     3) Ram (int)
     int ram = 16;
-    //     4) Screen size (float)
     float screenSize = 6.2f;
-    //     5) Amount of storage (int)
     int storageAmount = 32;
+
     
-    // 3 things it can do:
-    //     1) Make phone calls
     void makePhoneCall(std::string phoneNumber);
-    //     2) Take photos and videos
-    void takePhotoAndVideo(); 
-    //     3) Conenct to the internet
-    void connectToInternet(std::string wifiConnection);
+    void savePhotosAndVideos(int fileSize); 
+    void connectToInternet(bool connectedToWifi);
 };
 
+void Cellphone::makePhoneCall(std::string phoneNumber)
+{
+    if(phoneNumber == "Unavailable")
+    {
+        std::cout << "I'm sorry the phone number you are trying to reach is unavailable" << std::endl;
+    }
+    else
+    {
+        std::cout << "Connecting to call" << std::endl;
+    }
+}
+
+void Cellphone::savePhotosAndVideos(int fileSize)
+{
+    storageAmount -= fileSize;
+}
+
+void Cellphone::connectToInternet(bool connectedToWifi)
+{
+    if(connectedToWifi == false)
+    {
+        std::cout << "Please connect to a WiFi network." << std::endl;
+    }
+    else
+    {
+        std::cout << "Connected!" << std::endl;
+    }
+}
+
+
+// UDT 2
 struct MovieTheater
 {
-    // 5 properties: 
-    //     1) Number of theaters (int)
     int numTheaters = 13;
-    //     2) Cost of small popcorn (double)
     double costSmallPopcorn = 8.9;
-    //     3) Cost of one ticket (double)
-    double costOneTicket = 19.9;
-    //     4) Number of IMAX theaters (int)
+    float costOneTicket = 19.9f;
     int numImaxTheaters = 2;
-    //     5) Number of movies available (int)
     int numMoviesAvailable = 15;
-    // 3 things it can do:
-    //     1) Sell movie tickets
-    void sellMovieTicket(); 
-    //     2) Play movies
-    void playMovie();
-    //     3) Sell food and snacks
-    float sellFoodAndSnacks (std::string itemPurchased, float fullChargeAmount); // returns items purchasing and amount charged.
+    int seatsPerTheater = 60;
+    double cashInRegister = 3000.39;
+    
+    
+
+    void sellOutMovie(int seatsSold); 
+    bool useProjector(bool projectorIsWorking);
+    bool chargeCashForFoodAndSnacks(bool payingWithCash);
 };
 
+void MovieTheater::sellOutMovie(int seatsSold)
+{
+    if(seatsPerTheater == seatsSold)
+    {
+        std::cout << "This showing is sold out." << std::endl;
+    }
+    else
+    {
+        seatsPerTheater -= seatsSold + 1;
+        std::cout << "There are tickets avaialble!" << std::endl;
+    }
+}
+
+bool MovieTheater::useProjector(bool projectorIsWorking)
+{
+    if(projectorIsWorking == true)
+    {
+        return true;
+    }
+    else
+    {
+        numTheaters -= 1;
+        return false;
+    }
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise